### ISSUE TITLE
Error in secured post-multipart interface with large files

### ISSUE THREAD

**Issue by h3nn1n9** [quarkusio/quarkus#44929]
### Describe the bug

An error occurs in a Quarkus with a secured post-multipart interface. If a larger file (about 150MB) is uploaded, the system is stuck in an endless loop without ever reaching the first line in the 'uploadFile' method. If security is switched off or a small file is uploaded, the error does not occur.
The security dependecy is `io.quarkus:quarkus-elytron-security-properties-file`.

The following properties are set in application.properties:

> quarkus.http.auth.basic=true
quarkus.security.users.file.enabled=true
quarkus.security.users.file.users=users.properties
quarkus.security.users.file.roles=roles.properties
quarkus.security.users.file.plain-text=true
quarkus.http.limits.max-form-attribute-size=2000M
quarkus.http.limits.max-body-size=2000M

### Expected behavior

The file should be uploaded and a string with the file name should be returned.

### Actual behavior

If a larger file (about 150MB) is uploaded, the system is stuck in an endless loop without ever reaching the first line in the `uploadFile` method. If security is switched off or a small file is uploaded, the error does not occur.

### How to Reproduce?

[multipart-example.zip](https://github.com/user-attachments/files/18018948/multipart-example.zip)

To reproduce, a file with the name `bigFile.zip` must be placed in the `src/main/resources/` folder. The file is too large to upload to GitHub.

### Output of `java -version`

Temurin-21.0.5

### Quarkus version or git rev

3.16.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

apache-maven-3.9.9

**Comment by geoand** [comment]
Have you perhaps tried if the same occurs in using Quarkus REST?

**Comment by h3nn1n9** [comment]
Yes, I tested it with the small reproducer and the same error occurs using Quarkus REST.

**Comment by geoand** [comment]
Thanks. Can you upload that reproducer as well?

**Comment by h3nn1n9** [comment]
Here is a version of the reproducer with Quarkus REST.
[multipart-example-resteasy.zip](https://github.com/user-attachments/files/18024149/multipart-example-resteasy.zip)

**Comment by geoand** [comment]
🙏

**Comment by geoand** [comment]
If you want to check it today, go ahead. I was planning on having a look tomorrow

**Comment by michalvavrik** [comment]
> If you want to check it today, go ahead. I was planning on having a look tomorrow

Well, I'll check it this evening and if I can't tell, it will be waiting for you. I'll definitely update you in this issue with whatever I saw....

**Comment by geoand** [comment]
🙏🏽

**Comment by michalvavrik** [comment]
This issue is not specific for Quarkus Security or Quarkus REST or Quarkus RESTEasy. At any rate, I don't know how to fix it, but I'll provide context in an email and include you there @geoand . Thanks a million time @h3nn1n9 for reporting this. I suppose more info will follow by whoever will take care of this.

**Comment by michalvavrik** [comment]
Actually I found a fix, but I'll need to confirm it is the correct way to handle this. I'll send the email anyway.

**Comment by h3nn1n9** [comment]
Is there a reason why this bug is not included in the new LTS release 3.15.3? We always use the current LTS version and would therefore need the fix in this version.

**Comment by geoand** [comment]
cc @jmartisk

**Comment by jmartisk** [comment]
Hmm, it seems the `backport-3.15` label was added just at the last minute before the branch freeze and we didn't have any more triage meetings after that where we could notice it. But yeah, this probably should have been included :( Next time, when something important like this is added shortly before the freeze for the next LTS release, it would be nice to give me a ping to make sure we don't miss it. I will make sure it gets into 3.15.4...

**Comment by michalvavrik** [comment]
> Hmm, it seems the `backport-3.15` label was added just at the last minute before the branch freeze and we didn't have any more triage meetings after that where we could notice it. But yeah, this probably should have been included :( Next time, when something important like this is added shortly before the freeze for the next LTS maintenance release, it would be nice to give me a ping to make sure we don't miss it. I will make sure it gets into 3.15.4 and that I have one more look at the labeled issues just before the freeze, sorry for that...

I spoke about it with @rsvoboda right after this got merged and we agreed that it is a potentially sensitive change. I think little bake time won't hurt. It sounds awful, but even though we have tests, you can't test everything related to this change.

**Comment by jmartisk** [comment]
Ah, ok, good to hear. It is being baked in 3.17.6 so unless it proves problematic, we will add it to 3.15.4.

**Comment by geoand** [comment]
Makes sense, thanks!

**Comment by gsmet** [comment]
To clarify, the payload for 3.15.3 was frozen on January 7th per the new release process.

Now we might have to make exceptions from time to time but we need to be extremely careful about it as rushing a potentially sensitive patch can lead to a big catastrophe.

See https://github.com/quarkusio/quarkus/wiki/3.15-LTS-Release-Planning for more information.
